src/main/java/duke/task/EventTask.java

```diff
@@ -0,0 +1,36 @@
+package duke.task;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class EventTask extends Task {
+
+    private LocalDate date;
+
+    /**
+     * Constructor for EventTask object.
+     * @param name name of task.
+     * @param isDone whether or not task is done.
+     * @param date date of event.
+     */
+    public EventTask(String name, boolean isDone, String date) throws DateTimeException {
+        super(name, isDone);
+        this.date = LocalDate.parse(date);
+    }
+
+    public EventTask(String name, String date) throws DateTimeException {
+        super(name);
+        this.date = LocalDate.parse(date);
+    }
+
+    @Override
+    public String formatForFile() {
+        return "E" + super.formatForFile() + SAVE_DATA_MARKER + this.date + "\n";
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + this.date.format(DateTimeFormatter.ofPattern("d MMM yyyy")) + ")";
```

Comment:
Perhaps you could break this line into two lines as it is a bit long. Or maybe you could introduce a local variable for part of the string.